util/data_set_helpers.py

```diff
@@ -35,6 +39,80 @@ def dev(self):
     def test(self):
         return self._test
 
+class DataSet(object):
+    def __init__(self, filelist, thread_count, batch_size, numcep, numcontext, next_index=lambda x: x + 1):
+        self._coord = None
+        self._numcep = numcep
+        self._x = tf.placeholder(tf.float32, [None, numcep + (2 * numcep * numcontext)])
+        self._x_length = tf.placeholder(tf.int32, [])
+        self._y = tf.placeholder(tf.int32, [None,])
+        self._y_length = tf.placeholder(tf.int32, [])
+        self.example_queue = tf.PaddingFIFOQueue(shapes=[[None, numcep + (2 * numcep * numcontext)], [], [None,], []],
+                                                  dtypes=[tf.float32, tf.int32, tf.int32, tf.int32],
+                                                  capacity=2 * self._get_device_count() * batch_size)
+        self._enqueue_op = self.example_queue.enqueue([self._x, self._x_length, self._y, self._y_length])
+        self._close_op = self.example_queue.close(cancel_pending_enqueues=True)
+        self._filelist = filelist
+        self.batch_size = batch_size
+        self._numcontext = numcontext
+        self._thread_count = thread_count
+        self._files_list = self._create_files_list()
```

Comment:
Maybe name this `_list_of_files`? Having both the name `_filelist` and `_files_list` is a bit confusing.

Another idea would be to 
- Remove `_filelist` completely
- Pass `filelist` to `_create_files_list()`
- Use `_files_list` in `total_batches()`